libs/framework/src/service_tracker.c

```diff
@@ -163,26 +163,56 @@ celix_status_t serviceTracker_destroy(service_tracker_pt tracker) {
 
 celix_status_t serviceTracker_open(service_tracker_pt tracker) {
     celixThreadMutex_lock(&tracker->mutex);
-    bool alreadyOpen = tracker->open;
-    tracker->open = true;
+    bool addListener = false;
+    switch (tracker->state) {
+        case CELIX_SERVICE_TRACKER_CLOSED:
+            tracker->state = CELIX_SERVICE_TRACKER_OPENING;
+            addListener = true;
+            break;
+        case CELIX_SERVICE_TRACKER_CLOSING:
+            celix_bundleContext_log(tracker->context, CELIX_LOG_LEVEL_WARNING, "Cannot open closing tracker");
+            break;
+        default:
+            //nop
+            break;
+    }
     celixThreadMutex_unlock(&tracker->mutex);
 
-    if (!alreadyOpen) {
+    if (!addListener) {
```

Comment:
Ok, so if "addListener" is false (i.e. do not add listener), we add a listener ....
```suggestion
    if (noServiceListenerRegistered) {
```
And can we only add one service listener? I expect the "addListener" to be set to true once bundleConext_addServiceListener() has completed, so this is only done once (at least, that is what I expect).

Reply:
this was a bug. Icm with the wrong init of the initial state, this did not trigger test failures.
Updated this.